LiteFarmOrg/LiteFarm
# LF-2387/notify a user of tasks due today

Ticket [LF-2387](https://lite-farm.atlassian.net/browse/LF-2387)
Linked Ticket [LF-2386](https://lite-farm.atlassian.net/browse/LF-2386) and [PR](https://github.com/LiteFarmOrg/LiteFarm/pull/1912)

*It is best to review [this PR](https://github.com/LiteFarmOrg/LiteFarm/pull/1912) and merge it before reviewing or merging this PR

**To Test:** (same as Faizah's PR)
- Add this to the `packages/api/.env` file: 
  - `JWT_SCHEDULER_SECRET=Another_token_was_needed_for_the_scheduler`
- Run the backend tests in timeNotifcation.test.js, make sure they pass
- Run both the frontend and backend servers
- Create a couple users at the same farm, create a few tasks that are assigned and a few that are due today.
- Grab the id of the user that the tasks were assigned to from the "pg-litefarm" database.
- Use a tool such as [jwt.io](https://jwt.io/ ) to generate a JWT token with the same secret and payload:
  - `{ "requestTimedNotifications": true }` 
- Use a tool like Postman to send a request to:
  - `http://localhost:5001/time_notification/daily_due_today_tasks/<user_id>`
  - Make sure to add a Authorization Bearer token header with the JWT token you generated
- Once the request has been sent, go back to the frontend > click the bell icon in the top right corner > make sure there is a notification with the title 'Tasks due today'
- Click the notification > Click the 'Take Me There' button > make sure it takes you to the `/tasks` page and filters the tasks such that only tasks due today that are not completed yet show up (There should be three filters).
